src/modules/Address/index.tsx

```diff
@@ -100,7 +110,13 @@ class Index extends PureComponent<IndexProps, IndexState> {
   }
 
   render() {
-    const { addressData, balancesData } = this.state;
+    const { addressData, balancesData, accountStatus } = this.state;
+    const hash = this.props.computedMatch.params.hash;
+    if (accountStatus === undefined) {
+      return <Loading />;
+    } else if (accountStatus === 'nonexist') {
+      return <AddressNotFound address={hash} />;
+    }
```

Comment:
why add a redirect ?

you can render following componet directly:
```
<div>
	<Helmet>
	  <title>Address Not Onchain</title>
	</Helmet>
	<Grid className={classes.root} container justify="center">
	  <Grid item xs={12} md={8} lg={4}>
	    <Card className={classes.card}>
	      <Typography variant="h6" className={classes.headline}>
	        {t('account.address')} <strong>{address}</strong> {t('account.notOnchain')}
	      </Typography>
	    </Card>
	  </Grid>
	</Grid>
</div>
```

Reply:
It's necessary before the balance is returned.